### ISSUE TITLE
[Accessibility] Consider using different audio cues for action triggered and applied

### ISSUE THREAD

**Issue by jooyoungseo** [microsoft/vscode#233765]
Type: <b>Feature Request</b>

CC @meganrogge 

## Current Issue

"Code Action Request Triggered" and "Code Action Applied" use the same audio cues as "Chat Request Sent"and "Chat Response"

When you trigger either "Fix using Copilot" or "Explain using Copilot", you would hear "Chat Response" twice: the first one for action applied and the second one for when the Chat response arrives.

This is very confusing for those who rely on audio cues because the first "Chat Response" audio cue could signal that the response is ready to review, which is not.

Another issue that I have found when using this with the Copilot Vision extension is that you hear the "Chat Request Sent" audio cue as soon as you trigger "Refine alt text" action even before you type any query. This is because the action audio gets triggered when the action is fired. This may cause some cognitive confusion among blind people, making them misunderstand that any Chat request has been already sent.

### Option 1

You could use different audio cues for action triggered and applied from the ones already taken by Chat audio cues.

## Option 2

Or, you could mute the action audio cues when Chat-related actions are triggered and applied so to make the Chat audio cues are played only.

VS Code version: Code - Insiders 1.96.0-insider (399779942db4d7ab1bd6f6ae976482d0020f10ca, 2024-11-13T05:04:32.098Z)
OS version: Windows_NT x64 10.0.22631
Modes:

**Comment by meganrogge** [comment]
@jooyoungseo what do you think of using the voice recording stopped/started sounds for now?

Our sound designer is out at the moment, or would probably ask him for new ones.

**Comment by jooyoungseo** [comment]
@meganrogge voice recording stopped/started are better than Chat request sound ATM.

**Comment by meganrogge** [comment]
cc @afre100 we'll want to have unique sounds eventually

**Comment by meganrogge** [comment]
To verify:
- install the Copilot with Vision preview extension https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-copilot-vision
- enable screen reader mode
- enable the code action applied and code action triggered audio cues ![the quick pick for accessibility sounds is shown and filtered to just code action related ones, both are checked](https://github.com/user-attachments/assets/9c1954b7-f2ca-43c2-8e4f-a5548d0d3dde)
- in a markdown document, go to an image tag, trigger the code action to generate alt text
- verify you hear the sound for when it's triggered and when it's been applied

**Comment by alexr00** [comment]
I get "Invalid API key" when I try to give my Open AI API after clicking the alt text code action. Any other way to verify this?

**Comment by alexr00** [comment]
Hmm, looks like I hit this. I'll see if I can use my azure credit and get an API key there.

**Comment by alexr00** [comment]
Created https://github.com/microsoft/vscode-copilot-vision/pull/85 to spare anyone else that pain.

**Comment by AFre100** [comment]
@jooyoungseo @meganrogge Thanks for this feedback, all! I'll definitely look into this and keep you posted. Just fyi, i return from leave late Dec/early Jan (not early Dec, as previously thought)

**Comment by meganrogge** [comment]
Reopening so we get some unique sounds

**Comment by AFre100** [comment]
@meganrogge @jooyoungseo (et all). New sounds are always exciting — always open to exploring that — but I just wanted to zero in on a few things. 

The current flow is for the “progress” sound to loop while waiting for any Copilot related response. Would it not make sense for "Fix using Copilot" or "Explain using Copilot" to simply trigger the “progress” sound until Copilot returns a response? If I’m understanding it correctly, this would both 1) keep the audio cues consistent  and 2)mitigate cacophony 

With respect to needing a sound for “Code Action Applied,” do you think it’d make sense (and simplify things) to use the existing “Action Complete” sound?

**Comment by meganrogge** [comment]
@AFre100, yes, the flow already triggers that sound. 

Do you mean the `Save` sound or `Task Completed`?

**Comment by AFre100** [comment]
@meganrogge sorry for the confusion, yes, for "Code Action Applied" I was asking whether Task completed (aka action completed/success) sound would make sense, or if it needed a dedicated sound?

**Comment by AFre100** [comment]
@meganrogge @alexr00 would anyone be able to screen record (including sounds) the flow JooYoung described and send over? 

"When you trigger either "Fix using Copilot" or "Explain using Copilot", you would hear "Chat Response" twice: the first one for action applied and the second one for when the Chat response arrives."

**Comment by meganrogge** [comment]
we need a new sound so it's not confusing. 

@AFre100 the steps above should allow you to try it for yourself, which will probably help you to understand the flow

**Comment by AFre100** [comment]
@meganrogge, we discussed a global "success" sound at one point. Just wanted to confirm — what's the reason that the "Success" (aka Terminal Command Succeeded) sound isn't being floated here?

**Comment by AFre100** [comment]
@meganrogge et all, does this audio ux/flow sound right? 

-"Code Action Request Triggered" [new dedicated Sound 1] 
-(In contexts involving Copilot, e.g. "Fix using Copilot" /"Explain using Copilot” [existing “Progress” sound loop])
-"Code Action Applied" [new dedicated Sound 2]

**Comment by meganrogge** [comment]
Yes!

**Comment by AFre100** [comment]
Sounds good. Just wanted to confirm here for visibility—a new, dedicated audio cue is currently in development to replace the the pre-existing Sound.voiceRecordingStarted. 

(That reuse may pose future issues so it should be temporary. More soon—thanks!)
